src/Core/Managed/Shared/Extensibility/Implementation/TelemetryDebugWriter.cs

```diff
@@ -0,0 +1,78 @@
+﻿// <copyright file="TelemetryDebugWriter.cs" company="Microsoft">
+// Copyright © Microsoft. All Rights Reserved.
+// </copyright>
+
+#define DEBUG
+
+namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Diagnostics;
+    using Channel;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Platform;
+
+    /// <summary>
+    /// Writes telemetry items to debug output.
+    /// </summary>
+    public class TelemetryDebugWriter : IDebugOutput
```

Comment:
The are a lot of unit tests that actually test whether items are written to debug output. So I ended up reaching a compromise where the publicly exposed method is static, but we keep the interface for our unit tests.